src/classic/class/ReactClass.js

```diff
@@ -828,6 +828,12 @@ var ReactClass = {
       // This constructor is overridden by mocks. The argument is used
       // by mocks to assert on what gets mounted.
 
+      invariant(
+        this,
```

Comment:
In non-strict mode, won't `this` be `window`? Maybe check `this instanceof Constructor` instead?